src/sentry/tsdb/redis.py

```diff
@@ -265,6 +258,24 @@ def merge(self, model, destination, sources, timestamp=None):
                             ),
                         )
 
+    def delete(self, models, keys, start=None, end=None, timestamp=None):
+        rollups = self.get_active_series(start, end, timestamp)
+
+        with self.cluster.map() as client:
+            for rollup, series in rollups.items():
+                for timestamp in series:
+                    for model in models:
+                        for key in keys:
+                            model_key = self.get_model_key(key)
+                            client.hdel(
```

Comment:
is it worth trying to think "are we going to end up just deleting all keys in this hash?" and then just delete the whole toplevel hash key instead of each entry one by one, or is that not going to be happen much?